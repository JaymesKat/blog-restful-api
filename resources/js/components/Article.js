import React from 'react';

const Article = ({article}) => {

  const divStyle = {
      display: 'flex',
      flexDirection: 'column',
      margin: '30px 10px 10px 30px'
  }

  if(!article) {

    return(<div style={divStyle}><h2><strong>No Article is selected</strong></h2> </div>);
  }

  return(
    <div style={divStyle}>
      <h2><strong>{article.title}</strong></h2>
      <p> {article.body} </p>
      <h5> Post date : {article.created_at} </h5>

    </div>
  )
}

export default Article ;
