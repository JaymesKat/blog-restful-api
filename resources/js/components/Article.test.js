import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Article from './Article';

describe('Article', () => {
    const article = {
        "title": "Test Article",
        "body": "Sample text",
        "created_at": "2018-09-20 09:30:01"
    }

  it('should render correctly', () => {
    const output = shallow(
      <Article article={article} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
