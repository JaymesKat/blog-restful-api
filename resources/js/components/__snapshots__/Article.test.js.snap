// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "margin": "30px 10px 10px 30px",
    }
  }
>
  <h2>
    <strong>
      Test Article
    </strong>
  </h2>
  <p>
     
    Sample text
     
  </p>
  <h5>
     Post date : 
    2018-09-20 09:30:01
     
  </h5>
</div>
`;
